### PR TITLE
force reporter to update status

### PR DESCRIPTION
**Pull request by yuval-k**
Fixes a bug where the gateway may not updated the status on a resource.

added an integration test that reproduces the problem and now passes 

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3592

### CODE REVIEW

File: projects/gateway/pkg/syncer/translator_syncer.go [hunk 1]
```diff
@@ -305,8 +309,17 @@ func (s *statusSyncer) syncStatus(ctx context.Context) error {
 		currentStatuses := inputResourceBySubresourceStatuses[inputResource]
 		inputResource := resources.Clone(inputResource).(resources.InputResource)
 		reports := reporter.ResourceReports{inputResource: subresourceStatuses}
+		// set the last known status on the input resource.
+		// this may be different than the status on the snapshot, as the snapshot doesn't get updated
+		// on status changes.
+		if status, ok := localInputResourceLastStatus[inputResource]; ok {
+			inputResource.SetStatus(status)
+		}
```
Comment (kdorosh) [inline projects/gateway/pkg/syncer/translator_syncer.go:317]: debug logging if not ok?

Comment (yuval-k) [reply projects/gateway/pkg/syncer/translator_syncer.go:317]: not ok is expected as this map gets nuked in every new snapshot.
do you still think we should add debug log? i suppose better safe than sorry?

Comment (kdorosh) [reply projects/gateway/pkg/syncer/translator_syncer.go:317]: no i don't think so; i forgot the map gets nuked each time we set current proxies

File: projects/gateway/pkg/syncer/translator_syncer_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+		ts.Sync(ctx, snapshot())
+		// wait for proxy to be written
+		Eventually(func() (bool, error) {
+			_, err := proxyClient.Read("gloo-system", "gateway-proxy", clients.ReadOpts{})
+			if err != nil {
+				return false, err
+			}
+			return true, nil
+		}).Should(BeTrue())
```
Comment (kdorosh) [inline projects/gateway/pkg/syncer/translator_syncer_integration_test.go:174]: we can simplify this to:
```suggestion
		Eventually(func() (bool, error) {
			return proxyClient.Read("gloo-system", "gateway-proxy", clients.ReadOpts{})
		}).ShouldNot(BeNil())
```

File: projects/gateway/pkg/syncer/translator_syncer_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+			}
+			return proxy.Status.State, nil
+		})
+	}
+
+	AcceptProxy := func() {
+		proxy, err := proxyClient.Read("gloo-system", "gateway-proxy", clients.ReadOpts{})
+		Expect(err).NotTo(HaveOccurred())
+		proxy.Status = core.Status{State: core.Status_Accepted}
+		proxyClient.Write(proxy, clients.WriteOpts{OverwriteExisting: true})
```
Comment (kdorosh) [inline projects/gateway/pkg/syncer/translator_syncer_integration_test.go:161]: assert no error?

File: projects/gateway/pkg/syncer/translator_syncer_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+			return true, nil
+		}).Should(BeTrue())
+
+		// write the proxy status.
+		AcceptProxy()
+
+		// wait for the proxy status to be written in the VS
+		EventuallyProxyStatusInVs().Should(Equal(core.Status_Accepted))
+
+		// re-sync so now the snapshot, so that the snapshot has the updates status.
```
Comment (kdorosh) [inline projects/gateway/pkg/syncer/translator_syncer_integration_test.go:182]: nit, grammar here is confusing to read

**Review by kdorosh — COMMENTED** [review]
lgtm, just a couple nits

**Review by kdorosh — APPROVED** [review]

**Pull request merged**